src/app/features/de-duplication/bulk-merge-service.spec.ts

```diff
@@ -46,4 +47,23 @@ describe("BulkMergeService", () => {
 
     await expectEntitiesToBeInDatabase([mergedEntity], false, true);
   });
+
+  it("should merge related entities of recordB into recordA after merging", async () => {
+    const noteA = new Note();
+    noteA.relatedEntities = [recordA.getId()];
+
+    await entityMapper.save(noteA);
+
+    // mock merged entity
+    // Todo: this should be done in the service after updating executeMerge method
+    noteA.relatedEntities.push(recordB.getId());
+    await entityMapper.save(noteA);
+
+    const updatedNoteA = entityMapper.get(
+      Note.ENTITY_TYPE,
+      noteA.getId(),
+    ) as Note;
+    console.log(updatedNoteA);
+    expect(updatedNoteA.relatedEntities).toContain(recordB.getId());
```

Comment:
An additional test case can be: when `multiRelated = [recordA.getId(), recordB.getId()]` then only one time the merged ID should be included (not including it twice)